docs/developer/extending/webhooks/synchronous-events/stored-payment-method.mdx

```diff
@@ -186,3 +186,240 @@ The response in this case should have the following structure:
+
+## Process additional actions for payment method tokenization
+
+`PAYMENT_METHOD_PROCESS_TOKENIZATION_SESSION`
+
+:::info
+This feature was introduced in **Saleor 3.16**.
+:::
+
+The synchronous [PAYMENT_METHOD_PROCESS_TOKENIZATION_SESSION](api-reference/webhooks/enums/webhook-event-type-sync-enum.mdx#code-style-fontweight-normal-webhookeventtypesyncenumbpayment_method_process_tokenization_sessionbcode) webhook is triggered when a customer requests to process additional actions required for tokenizing the payment method. This webhook is activated by executing the `paymentMethodProcessTokenization` mutation.
+
+The webhook payload includes details about the [customer](api-reference/users/objects/user.mdx), [channel](api-reference/channels/objects/channel.mdx), and [data]. The `data` field contains the JSON data provided as input in the [paymentMethodProcessTokenization](api-reference/payments/mutations/payment-method-process-tokenization.mdx) mutation.
+
+The response must adhere to the expected format and will be returned within the [paymentMethodProcessTokenization](api-reference/payments/mutations/payment-method-process-tokenization.mdx) mutation.
+
+### Request
+
+Saleor will send a [PAYMENT_METHOD_PROCESS_TOKENIZATION_SESSION](api-reference/webhooks/enums/webhook-event-type-sync-enum.mdx#code-style-fontweight-normal-webhookeventtypesyncenumbpayment_method_process_tokenization_sessionbcode) webhook using the [PaymentMethodProcessTokenizationSession](api-reference/payments/mutations/payment-method-initialize-tokenization.mdx) subscription type or with a predefined payload in case of a missing subscription query.
+
+For detailed information on building webhook subscription queries, please refer to [this documentation](../../../extending/webhooks/subscription-webhook-payloads).
+
+The following example demonstrates a sample webhook subscription defined to handle the tokenization of a payment method:
+
+```graphql
+subscription {
+  event {
+    ... on PaymentMethodProcessTokenizationSession {
+      id
+      user {
+        id
+      }
+      channel {
+        id
+      }
+      data
+    }
+  }
+}
+```
+
+The next example illustrates the predefined payload that will be used if a subscription query is not provided:
+
+```json
+{
+  "user_id": "VXNlcjoyOA==",
+  "channel_slug": "main",
+  "data": { "data": "from-input" }
+}
+```
+
+### Response
+
+- A response indicating successful tokenization should include the `result`, the `id` of the stored payment method, and optionally `data` to be passed to the storefront:
+
+```json
+{
+  "result": "SUCCESSFULLY_TOKENIZED",
+  "id": "payment-method-id",
+  "data": { "data": "additional-data" }
+}
+```
+
```

Comment:
Same as in PAYMENT_METHOD_INITIALIZE_TOKENIZATION_SESSION, we should have 4th result here for pending tokenization

Reply:
Added